stories/modal.stories.js

```diff
@@ -138,3 +138,81 @@ export const Normal = () => {
     </>
   )
 }
+
+export const CustomClose = () => {
+  const modalClasses = Object.freeze({
+    footer: 'flex justify-between',
+  })
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [size, setSize] = useState('md')
+
+  const inputProps = { placeholder: 'Placeholder text' }
+
+  const onClose = () => {
+    setIsOpen(true)
+  }
+
+  const onClick = (s) => {
+    setIsOpen(!isOpen) 
+    setSize(s)
+  }
+
+  const footerContent = (
+    <>
+      {size === 'sm' ? 
+        <>
+          <Button size={size === 'lg' ? 'lg' : 'md'} onClick={() => onClick('lg')}>Cancel</Button>
+          <Button variant='filled' size={size === 'lg' ? 'lg' : 'md'}>Submit</Button>
+        </>
+        :
+        <>
+          <Button size={size === 'lg' ? 'lg' : 'md'} type='error' onClick={() => onClick('lg')}>Cancel</Button>
+          <div className="right-button flex flex-row">
+            <div className="pr-4">
+              <Button variant='outlined' size={size === 'lg' ? 'lg' : 'md'}>Reset</Button>
+            </div>
+            <Button variant='filled' size={size === 'lg' ? 'lg' : 'md'}>Save Selection</Button>
+          </div>
+        </>
+      }
+    </>
+  )
+  
+  return (
+    <>
+      <Modal 
+        size={size} 
+        open={isOpen} 
+        closeModal={onClose} 
```

Comment:
In the case of handling show/hide of the header close icon with presence of `closeModal` or not, here `onClose` can be handled within the `onClick` handler of the `cancel` button, instead of passing it via `closeModal`